### PR TITLE
UEFI

### PR DESCRIPTION
**Pull request by N00byEdge**

### CODE REVIEW

File: src/platform/uefi_aarch64/main.zig [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+    if (.Success != uefi.system_table.boot_services.?.locateProtocol(&guid, null, @ptrCast(*?*c_void, &ptr))) {
+        return null;
+    }
+    return ptr;
+}
+
+pub fn handleProtocol(handle: uefi.Handle, comptime T: type) callconv(.Inline) ?*T {
+    var ptr: *T = undefined;
+    const guid: std.os.uefi.Guid align(8) = T.guid;
+    if (.Success != uefi.system_table.boot_services.?.handleProtocol(handle, &guid, @ptrCast(*?*c_void, &ptr))) {
```
Comment (StaticSaga) [inline src/platform/uefi_aarch64/main.zig]: Why is the order inverted here?

File: src/platform/uefi_aarch64/main.zig [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+    sabaton.add_tag(@ptrCast(*sabaton.Stivale2tag, stivale2_memmap_bytes.ptr));
+
+    uefiVital(uefi.system_table.boot_services.?.exitBootServices(uefi.handle, memmap.key), "Exiting boot services");
+
+    conout = null; // We can't call UEFI after exiting boot services
+
+    sabaton.paging.apply_paging(&paging_root);
+
+    asm volatile (
+        \\  MSR SPSel, XZR
```
Comment (StaticSaga) [inline src/platform/uefi_aarch64/main.zig]: Maybe refactor this into a separate function

File: uefidir/image/startup.nsh [hunk 3]
```diff
@@ -0,0 +1 @@
+FS0:\EFI\BOOT\BOOTA64.EFI
```
Comment (StaticSaga) [inline uefidir/image/startup.nsh]: Why is startup.nsh necessary here? It should autoboot EFI\BOOT\BOOTA64.EFI.

**Review by StaticSaga — COMMENTED** [review]
Looks good to me

File: src/platform/uefi_aarch64/main.zig (hunk 1 above)
Comment (N00byEdge) [reply src/platform/uefi_aarch64/main.zig]: Because where I stole it from had it in that order :^)

File: uefidir/image/startup.nsh (hunk 3 above)
Comment (N00byEdge) [reply uefidir/image/startup.nsh]: It doesn't autoboot it for me though, and I'm not sure why. Are my efi vars just wrong?

File: src/platform/uefi_aarch64/main.zig (hunk 2 above)
Comment (N00byEdge) [reply src/platform/uefi_aarch64/main.zig]: Fair

File: uefidir/image/startup.nsh (hunk 3 above)
Comment (StaticSaga) [reply uefidir/image/startup.nsh]: Very likely, but I am not sure how or why. I had the same issue with x86 uefi and it somehow resolved with a totally unrelated change

**Review by StaticSaga — COMMENTED** [review]
Maybe change the build mode to ReleaseSmall to be consistent.

File: uefidir/image/startup.nsh (hunk 3 above)
Comment (N00byEdge) [reply uefidir/image/startup.nsh]: Can confirm, fixing my boot settings makes it work.

**Pull request merged**